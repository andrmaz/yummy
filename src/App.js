import React, { Component } from 'react';
import { Container} from '@chakra-ui/react';
import Header from './components/Header';
import FormFields from './components/FormFields';
import ListItems from './components/ListItems';
import './App.css';

class App extends Component {
  constructor(props) {
    super(props);
    this.onSearch = this.onSearch.bind(this);
    this.state = { results : null };
  }

  onSearch(data) {
    this.setState({ results: data });
  };

  render(){
    const { results } = this.state;
    return (
      <Container h='100vh' maxW='1200px' mb='10vh'>
        <Header />
        <FormFields
          onSearch={this.onSearch}
        />
        <ListItems 
          results={results}
        />
      </Container>
    );
  };
};

export default App;
