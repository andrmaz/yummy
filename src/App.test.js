import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

let wrapper;
const mockState = { count: 2, hits: [1, 2] };

describe('App Component', () => {
  beforeEach(() => {
    wrapper = shallow(<App />);
    expect.hasAssertions();
  });
  it('should render Component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should have null state', () => {
    //Accessing component state
    expect(wrapper.state('results')).toEqual(null);
  });
  it('should pass object to state correctly', () => {
    wrapper.setState({ results: mockState});
    expect(wrapper.state('results')).toEqual(mockState);
    expect(wrapper.state('results')).toHaveProperty('count', 2);
  });
  it('shallow wrapper instance should not be null', () => {
    const instance = wrapper.instance();
    expect(instance).toBeInstanceOf(App);
  });
  it('should set state correctly', () => {
    wrapper.instance().onSearch(mockState);
    expect(wrapper.state('results')).toEqual(mockState);
  });
});
