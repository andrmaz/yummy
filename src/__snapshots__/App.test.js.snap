// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should render Component correctly 1`] = `
<Container
  h="100vh"
  maxW="1200px"
  mb="10vh"
>
  <Header />
  <SearchInputs
    onSearch={[Function]}
  />
  <Memo(ListItems)
    results={null}
  />
</Container>
`;
